### PR TITLE
remove some sync calls

### PR DESCRIPTION
**Pull request by raymondng**
WIP; only dealt with a few so far; more to come

Some "async" are brute force -- e.g. HttpFetcher.get -- a real change will require a HTTP client change, which is under investigation

Most of the renaming has no significance -- just a way to help with refactoring

@ymatsuda @yingjieMiao @eishay @leogrim

### CODE REVIEW

File: kifi-backend/modules/scraper/app/com/keepit/scraper/ScrapeWorker.scala [hunk 1]
```diff
@@ -228,35 +230,39 @@ class ScrapeWorkerImpl @Inject() (
...
-      case NotModified => handleNotModified(latestUri.url, info)
-      case error: Error => handleScrapeError(latestUri, error, info)
+  private def process(uri: NormalizedURI, info: ScrapeInfo, pageInfoOpt: Option[PageInfo], proxyOpt: Option[HttpProxy]): Future[Option[Article]] = {
+    for {
+      fetchedArticle <- safeFetch(uri, info, proxyOpt)
+      uriOpt <- dbHelper.getNormalizedUri(uri)
+    } yield {
+      val articleOpt = uriOpt flatMap { latestUri =>
+        callEmbedly(latestUri) // @martin: do we need to do this for every scrape?
+
```
Comment (yingjieMiao) [inline kifi-backend/modules/scraper/app/com/keepit/scraper/ScrapeWorker.scala:240]: I was discussing this with you back then. If you trace the function deeper, you will see we only call embedly if the embedly info is missing or is "old enough"

Comment (raymondng) [inline kifi-backend/modules/scraper/app/com/keepit/scraper/ScrapeWorker.scala:240]: Oh I see.  Thanks for the clarifications. 
Indeed I didn't trace it all the way down -- my bad. Maybe rename this method to make it more obvious what's going on?

Comment (yingjieMiao) [inline kifi-backend/modules/scraper/app/com/keepit/scraper/ScrapeWorker.scala:240]: yep. maybe `callEmbedlyIfNecessary` ?

Comment (raymondng) [inline kifi-backend/modules/scraper/app/com/keepit/scraper/ScrapeWorker.scala:240]: :+1:

**Comment by yingjieMiao** [comment]
lgtm

**Pull request merged**